pkg/specgen/generate/oci.go

```diff
@@ -7,71 +7,21 @@ import (
 	"github.com/containers/common/libimage"
 	"github.com/containers/common/pkg/config"
 	"github.com/containers/podman/v4/libpod/define"
-	"github.com/containers/podman/v4/pkg/rootless"
 	"github.com/containers/podman/v4/pkg/specgen"
+	"github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/opencontainers/runtime-tools/generate"
-	"github.com/sirupsen/logrus"
-	"golang.org/x/sys/unix"
 )
 
 func addRlimits(s *specgen.SpecGenerator, g *generate.Generator) {
-	var (
-		isRootless = rootless.IsRootless()
-		nofileSet  = false
-		nprocSet   = false
-	)
-
-	if s.Rlimits == nil {
-		g.Config.Process.Rlimits = nil
-		return
-	}
+	g.Config.Process.Rlimits = nil
 
 	for _, u := range s.Rlimits {
 		name := "RLIMIT_" + strings.ToUpper(u.Type)
-		if name == "RLIMIT_NOFILE" {
-			nofileSet = true
-		} else if name == "RLIMIT_NPROC" {
-			nprocSet = true
-		}
-		g.AddProcessRlimits(name, u.Hard, u.Soft)
-	}
-
-	// If not explicitly overridden by the user, default number of open
-	// files and number of processes to the maximum they can be set to
-	// (without overriding a sysctl)
-	if !nofileSet {
```

Comment:
I am pretty sure we actually want this. The problem is that we only set it once on create instead of for each start.

So I think this must be moved into `func (c *Container) generateSpec()` instead

Reply:
Is there a problem with leaving the ulimits in the spec file empty? The container process should inherit the limits, right?

Is that generateSpec invoked on each run and the spec passed to crun, or is it also saved on disk for the next time the container is started?

Reply:
Ok, so `func (c *Container) generateSpec()` from container_internal_common.go gets executed each time the container is started if it's in `ContainerStateConfigured` or `ContainerStateExited`. I'll put setting the max ulimits in there and see if that fixes the CI issues for rootless.

I'll also disable the new test for rootfull, as this issue is not really applicable to rootfull.

Reply:
Looks like this approach works on my end.

Reply:
Incorporated into latest version of this PR